### PR TITLE
core: Enforce SEP-31 sell limits in SEP-38 quote calls

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Enforce SEP-31 aset sell limits in SEP-38 quote calls

### Why

The expected behavior for invalid requests should be a 400

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java [hunk 1]
```diff
@@ -187,6 +189,17 @@ public GetPriceResponse getPrice(Sep38GetPriceRequest getPriceRequest) throws An
       throw new BadRequestException("Unsupported context. Should be one of [sep6, sep31].");
     }
 
+    // Check SEP31 send limits
+    String[] assetCode = sellAsset.getAsset().split(":");
+    AssetInfo asset = assetService.getAsset(assetCode[1]);
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java:194]: The asset can be null.

File: core/src/main/java/org/stellar/anchor/util/SepHelper.java [hunk 2]
```diff
@@ -70,6 +71,20 @@ public static void validateAmount(String messagePrefix, String amount) throws An
...
+  public static void validateAmountLimit(String messagePrefix, String amount, Long min, Long max)
+      throws AnchorException {
+    BigDecimal sAmount = validateAmount("", amount);
+    BigDecimal bdMin = new BigDecimal(min);
+    BigDecimal bdMax = new BigDecimal(max);
+    if (sAmount.compareTo(bdMin) == -1) {
+      throw new BadRequestException(messagePrefix + "amount less than min limit");
+    }
+    if (sAmount.compareTo(bdMax) == 1) {
+      throw new BadRequestException(messagePrefix + "amount exceeds max limit");
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/util/SepHelper.java:86]: 💡 We may consider using `String.format()` to format a Java String.

Comment (erika-sdf) [reply core/src/main/java/org/stellar/anchor/util/SepHelper.java]: Changed this to use String.format

File: core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java (hunk 1 above)
Comment (erika-sdf) [reply core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java:197]: Added a check for asset being not null

**Review by lijamie98 — APPROVED** [review]
LGTM.

**Pull request merged**